### PR TITLE
[CBRD-24337] Users cannot create tables and views with the system class name.

### PR DESCRIPTION
**Pull request by youngjinj**
http://jira.cubrid.org/browse/CBRD-24337

Users cannot create tables and views with the system class name.
Users cannot rename the system class name.

### CODE REVIEW

**Review by ctshim — APPROVED** [review]

**Review by hyunikn — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

File: src/parser/parser_support.c [hunk 1]
```diff
@@ -10196,70 +10196,106 @@ pt_set_user_specified_name (PARSER_CONTEXT * parser, PT_NODE * node, void *arg,
...
+	PT_SYNONYM_OWNER_NAME (node) = pt_name (parser, pt_get_qualifier_name (parser, PT_SYNONYM_NAME (node)));
+	if (pt_get_qualifier_name (parser, PT_SYNONYM_TARGET_NAME (node)) == NULL
+	    && sm_check_system_class_by_name (PT_NAME_ORIGINAL (PT_SYNONYM_TARGET_NAME (node))) == true)
+	  {
+	    PT_SYNONYM_TARGET_OWNER_NAME (node) = pt_name (parser, "dba");
+	  }
+	else
+	  {
+	    PT_SYNONYM_TARGET_OWNER_NAME (node) =
+	      pt_name (parser, pt_get_qualifier_name (parser, PT_SYNONYM_TARGET_NAME (node)));
```
Comment (hyunikn) [inline src/parser/parser_support.c:10260]: Is it possible that pt_get_qualifier_name (parser, PT_SYNONYM_TARGET_NAME (node)) == NULL at this line? If so, isn't it a problem?

File: src/parser/parser_support.c [hunk 2]
```diff
@@ -10196,70 +10196,106 @@ pt_set_user_specified_name (PARSER_CONTEXT * parser, PT_NODE * node, void *arg,
...
 	  else
 	    {
-	      PT_SYNONYM_TARGET_OWNER_NAME (node) =
-		pt_name (parser, pt_get_qualifier_name (parser, PT_SYNONYM_TARGET_NAME (node)));
+	      assert (PT_IS_NAME_NODE (node->info.expr.arg1));
+
+	      PT_NODE *name = node->info.expr.arg1;
+
+	      original_name = name->info.name.original;
+	      resolved_name = name->info.name.resolved;
```
Comment (hyunikn) [inline src/parser/parser_support.c]: [Q] is resolved_name always NULL at this line?

**Review by hyunikn — APPROVED** [review]

File: src/parser/parser_support.c (hunk 2 above)
Comment (youngjinj) [reply src/parser/parser_support.c]: Don't set name->info.name.resolved before. So it's probably NULL.
It seems to be left over from the old code.

I will check more and remove if unnecessary.

File: src/parser/parser_support.c (hunk 1 above)
Comment (youngjinj) [reply src/parser/parser_support.c:10260]: This is code that cannot be NULL.
I'm going to add an assert to check for NULL.

**Pull request merged**